### ISSUE TITLE
Support for GraphQL and headless mode?

### ISSUE THREAD

**Issue by MatthiasGrandl** [ethercreative/seo#255]
Any plans to add support for the new GraphQL and headless modes? What would be needed to get the focus keywords to run in headless?

**Comment by LarsDol** [comment]
Out-of-the-box GraphQL support would be great!

I'm currently writing a nuxt plugin to parse the data generated by the SEO plugin. It will pass the proper data to nuxt's head function. I've added a `seo` field (the plugin's SEO field) to an entry.

GraphQL query:
```
{
  entries(section: "homepage", limit: 1) {
    id,
    title,
    ... on homepage_homepage_Entry {
      seo
    }
  }
}
```

This returns the following error: `Expected a value of type \"String\" but received: instance of ether\\seo\\models\\data\\SeoData`.

Any known workarounds for this? Or a planning on supporting GraphQL would be great!

Thanks!

**Comment by FreekVR** [comment]
To anyone watching this, I've forked the repo and added some experimental support for all the Seo fields to the Craft GQL implementation. I wasn't able to test this in a production project yet, but in my experiments it worked well :)

https://github.com/FreekVR/seo

Would appreciate if other people would help out with testing, I'll then create a PR to this repo.

You can install it by adding the following to your composer.json:

```json
"repositories": [
        {
            "type": "vcs",
            "url": "https://github.com/FreekVR/seo.git"
        }
    ]
```

and change your dependency for the seo-plugin to
`"ether/seo": "v3.x-dev"`

Here's an example gql query that will get most of the SEO fields:

```gql
{
    entries(
        section: "homepage",
        limit: 1
    ) {
        id,
        title,
        ... on homepage_homepage_Entry {
            seo { 
              title, description, 
              keywords { keyword, rating },
              social { 
                twitter { title, image { url }, description } 
                facebook { title, image { url }, description } 
              },
              advanced { canonical, robots }
            }
        } 
    }
}
```

**Comment by jamesedmonston** [comment]
We've merged in @FreekVR's changes and released `3.6.6`. Thanks!
